### PR TITLE
Kernel/Memory: Clang-Tidy fixes

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
This part of the code base was surprisingly clean...

Main thing to check is the logic in the second commit.

### CODE REVIEW

File: Kernel/Memory/MemoryManager.cpp [hunk 1]
```diff
@@ -198,14 +198,11 @@ bool MemoryManager::is_allowed_to_mmap_to_userspace(PhysicalAddress start_addres
...
     auto end_address = start_address.offset(range.size());
-    for (auto& current_range : m_reserved_memory_ranges) {
-        if (current_range.start > start_address)
-            continue;
-        if (current_range.start.offset(current_range.length) < end_address)
-            continue;
-        return true;
-    }
-    return false;
+    return AK::any_of(
```
Comment (bgianfo) [inline Kernel/Memory/MemoryManager.cpp:201]: What is the motivation for converting loops to `AK::any_of`, other than to appease an arbitrary clang-tidy rule? I don't personally find this better / easier to read.

Comment (Hendiadyoin1) [reply Kernel/Memory/MemoryManager.cpp:201]: None really,
mainly to appease clang-tidy
I can drop it if you want

Comment (bgianfo) [reply Kernel/Memory/MemoryManager.cpp:201]: IMHO we should drop those changes, and disable that rule if this is all it flags.

Comment (awesomekling) [reply Kernel/Memory/MemoryManager.cpp:201]: Yeah, let's drop those changes and disable the rule.

**Review by awesomekling — CHANGES_REQUESTED** [review]

**Pull request merged**